docs/prover/diagnosis/index.md

```diff
@@ -5,15 +5,238 @@ Diagnostic Tools
+:name: timeout tac report explanation box
+:align: center
+
+Timeout TAC report with expanded statistics- and explanation-box
+```
+
+The above picture shows the expanded statistics- and explanation box. Currently,
+the contents of the box are the following:
+ - explanations for each color used in the CFG nodes
+ - general difficulty-related statistics pertaining to the whole program
+ - per-call breakdown of the "path count" statistic
+ - per-call breakdown of the "number of nonlinear operations" statistic
+
+
+#### Split- and heuristic difficulty-coloring in CFG and source code
+
+The fill-colors of the graph nodes in the timeout TAC report summarize the
+outcomes of the individual control flow splits for the current rule. See
+{ref}`control-flow-splitting` for background on the procedure.
+An additional violet gradient is used to indicate nodes that are likely to 
+contain code that is difficult for the Prover to handle.
+
+Nodes that have a green fill color are contained exclusively in splits that have
+been successfully proven correct by the current run of the Prover. This means
+they are at least manageable for the Prover, although they still might have
+taken significant time to prove.
+
+Nodes that have a orange fill color lie on a split that whose checking has timed
+out under the timeout specified by {ref}`--smt_timeout`. This means that the
+combination of the orange nodes was too difficulty to solve within that time.
+Note that this does not say anything about the difficulty of any individual
+orange node.
+
+```{figure} timeout-tac-report-green-nodes.png
+:name: timeout tac report orange and green nodes
+:align: center
+
+CFG segment with orange and green nodes
+```
+
+Nodes that have a yellow fill color are on splits that were still queued for
+solving when the timeout on the orange nodes happened. No strong statements can
+be made about their difficulty at this point.
+
+```{figure} timeout-tac-report-yellow-nodes.png
+:name: timeout tac report yellow nodes
+:align: center
+
+CFG segment with nodes that have not yet been queued during control flow splitting,
+indicated by the yellow fill color
+```
+
+When a call node has a violet gradient, this means that its subgraph contains at
+least one node with a violet gradient. When a regular node has a violet gradient
+this means that our heuristic analysis of the node's difficulty has marked it as
+potentially difficult. This typically means that there are at least two
+nonlinear operations in the node, or a high number of case splits. When a
+regular node is marked violet, the source code in the source code box of the
+report will contain highlights of the relevant lines, as well as a summary of
+the difficult operations a the top of the block corresponding to the node.
+
+```{figure} timeout-tac-report-source-code.png
+:name: timeout tac report source code highlights 
+:align: center
+
+Source code with highlighting and difficulty summary
```

Comment:
Maybe explain a little more, e.g. contains one multiplication and one division operation...

Reply:
fair point, expanded a little on the nonlinear and the case splits